### PR TITLE
Added binding for GEORADIUSBYMEMBER

### PR DESCRIPTION
**Pull request by icha024**
Change-Id: Ia6144617f42629af4c022e595c444ddc6d66f1a3

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -1708,3 +1714,31 @@ func (c *commandable) GeoRadius(query *GeoRadiusQuery) *GeoLocationCmd {
 	c.Process(cmd)
 	return cmd
 }
+
+func (c *commandable) GeoDist(key string, geo1, geo2 *GeoLocation, unit string) *FloatCmd {
```
Comment (vmihailenco) [inline commands.go]: Please rename `WithDistance` to `WithDist` to be consistent.

File: commands.go [hunk 2]
```diff
@@ -1708,3 +1714,31 @@ func (c *commandable) GeoRadius(query *GeoRadiusQuery) *GeoLocationCmd {
...
+		unit = "km"
+	}
+	args := make([]interface{}, 5)
+	args[0] = "GEODIST"
+	args[1] = key
+	args[2] = geo1.Name
+	args[3] = geo2.Name
+	args[4] = unit
+
+	cmd := NewFloatCmd(args...)
```
Comment (vmihailenco) [inline commands.go]: `NewFloatCmd("GEODIST", key, geo1.Name, geo2.Name, unit)`

File: commands.go (hunk 1 above)
Comment (vmihailenco) [inline commands.go]: And replace `geo1, geo2 *GeoLocation` with `member1, member2 string`.

File: commands.go [hunk 3]
```diff
@@ -1708,3 +1714,31 @@ func (c *commandable) GeoRadius(query *GeoRadiusQuery) *GeoLocationCmd {
...
+}
+
+func (c *commandable) GeoHash(key string, geo1, geo2 *GeoLocation) *StringSliceCmd {
+	args := make([]interface{}, 4)
+	args[0] = "GEOHASH"
+	args[1] = key
+	args[2] = geo1.Name
+	args[3] = geo2.Name
+
+	cmd := NewStringSliceCmd(args...)
```
Comment (vmihailenco) [inline commands.go]: `NewStringSliceCmd("GEOHASH", key, name1, name2)`

File: commands.go [hunk 4]
```diff
@@ -1708,3 +1714,31 @@ func (c *commandable) GeoRadius(query *GeoRadiusQuery) *GeoLocationCmd {
...
+	args[2] = geo1.Name
+	args[3] = geo2.Name
+	args[4] = unit
+
+	cmd := NewFloatCmd(args...)
+	c.Process(cmd)
+	return cmd
+}
+
+func (c *commandable) GeoHash(key string, geo1, geo2 *GeoLocation) *StringSliceCmd {
```
Comment (vmihailenco) [inline commands.go]: Replace `geo1, geo2 *GeoLocation` with `key string, members ...string`

File: commands.go [hunk 5]
```diff
@@ -1677,16 +1677,22 @@ func (c *commandable) GeoAdd(key string, geoLocation ...*GeoLocation) *IntCmd {
 }
 
 func (c *commandable) GeoRadius(query *GeoRadiusQuery) *GeoLocationCmd {
```
Comment (vmihailenco) [inline commands.go]: I think signature here should be:

```
func (c *commandable) GeoRadius(key, longitude, latitude, query *GeoRadiusQuery) *GeoLocationCmd
func (c *commandable) GeoRadiusByMember(key, member, query *GeoRadiusQuery) *GeoLocationCmd
```

and underneath you can use some general function. Feel free to leave this change for me.

File: commands.go (hunk 1 above)
Comment (icha024) [inline commands.go]: Done.

File: commands.go (hunk 2 above)
Comment (icha024) [inline commands.go]: Done.

File: commands.go (hunk 3 above)
Comment (icha024) [inline commands.go]: Done.

File: commands.go (hunk 4 above)
Comment (icha024) [inline commands.go]: Done.

File: commands.go (hunk 5 above)
Comment (icha024) [inline commands.go]: I've exposed 2 different methods now, but am undecided about the new method signature - one looks cleaner but the other aligns more with Redis itself....

**Comment by vmihailenco** [comment]
Thanks for your work. I created https://github.com/go-redis/redis/pull/176 with some edits. Please let me know if you spot any mistakes.

**Comment by vmihailenco** [comment]
https://github.com/go-redis/redis/pull/176 was merged, closing this one. @icha024 , thanks again.

**Pull request closed without merging**